### ISSUE TITLE
Failure creating vertex with name close to MAX_LABEL_NAME_LEN 65535

### ISSUE THREAD

**Issue by MironAtHome** [apache/age#1986]
The maximum langth for label ( vertex ) name for application Age is defined as following:
`#define MAX_LABEL_NAME_LEN 65535`
in the file name_validation.h
Checking behavior of application, creating vertexs with length approaching limit of 65535 characters encountered this behavior:
1 Attempt to create first edge with name length 65536 success.
However 
2 subsequent request to create vlabel with name with last two characters removed fails with message:
-- ERROR:  label "vertexwithextremelyveryveryveryverylongnamenamenamenamenamename" already exists 
-- SQL state: 3F000
I think 65535 is not very reasonable length for the name of vertex. So, please do reconsider, but if it is decided to support 65535 character legth, please do ensure it is working.
```
select create_vlabel('test_graph', 'vertexwithextremelyveryveryveryverylongnamenamenamename<...>');
-- NOTICE:  VLabel "vertexwithextremelyveryveryveryverylongnamenamenamenamenamename" has been created
-- Successfully run. Total query runtime: 3 secs 93 msec.
-- 1 rows affected.

select create_vlabel('test_graph', 'vertexwithextremelyveryveryveryverylongnamenamenamename<...>');
-- ERROR:  label "vertexwithextremelyveryveryveryverylongnamenamenamenamenamename" already exists 
-- SQL state: 3F000

```

**Comment by jrgemignani** [comment]
@MironAtHome Good find! 

This one is due to the usage of the PG type `Name`, see below. It looks like we'll either have to limit names to `NAMEDATALEN` or use a different method.

It appears that whomever used `Name` in this case either didn't realize the drawbacks or didn't expect it to ever be longer than `NAMEDATALEN`.

```
/*
 * Maximum length for identifiers (e.g. table names, column names,
 * function names).  Names actually are limited to one fewer byte than this,
 * because the length must include a trailing zero byte.
 *
 * Changing this requires an initdb.
 */
#define NAMEDATALEN 64
```

```
/*
 * Representation of a Name: effectively just a C string, but null-padded to
 * exactly NAMEDATALEN bytes.  The use of a struct is historical.
 */
typedef struct nameData
{
        char            data[NAMEDATALEN];
} NameData;
typedef NameData *Name;
```

**Comment by jrgemignani** [comment]
@MironAtHome I'm going to need to check the whole data path to see which fix would be the best for this one.

**Comment by jrgemignani** [comment]
@MironAtHome It appears that if the PG type Name is used, then it will always be truncated at `NAMEDATALEN` on input -

```
Datum
namein(PG_FUNCTION_ARGS)
{
        char       *s = PG_GETARG_CSTRING(0);
        Name            result;
        int                     len;

        len = strlen(s);

        /* Truncate oversize input */
        if (len >= NAMEDATALEN)
                len = pg_mbcliplen(s, len, NAMEDATALEN - 1);

        /* We use palloc0 here to ensure result is zero-padded */
        result = (Name) palloc0(NAMEDATALEN);
        memcpy(NameStr(*result), s, len);

        PG_RETURN_NAME(result);
}
```

This also means that there won't be any warning if it exceeds `NAMEDATALEN` :(

**Comment by jrgemignani** [comment]
@MironAtHome I'm working on a fix. Hopefully, it will be done tomorrow.

**Comment by jrgemignani** [comment]
@MironAtHome Unfortunately, this one can't be extended to 65535 due to naming constraints elsewhere. Basically, a label is a relation and relation names are constrained to `NAMEDATALEN` in length. There's that constant again. 

I found this out when I tried creating labels with the new lengths. So, what I will need to do is create a constraint on label names to be less than (`NAMEDATALEN` - 2). This will allow us to catch names that are too long, because `namein` truncates then beyond that.
